msal/application.py

```diff
@@ -194,6 +194,21 @@ def obtain_token_by_username_password(self, username, password, **kwargs):
             username, password, headers=headers, **kwargs)
 
 
+def _msal_extension_check():
+    # Can't run this in module or class level otherwise you'll get circular import error
+    try:
+        from msal_extensions import __version__ as v
+        major, minor, _ = v.split(".", maxsplit=3)
```

Comment:
Just as an FYI, Azure CLI uses [`packaging.version`](https://packaging.pypa.io/en/latest/version.html) for version parsing:

https://github.com/Azure/azure-cli/blob/b6ab754b68efba60b30806fa80dc3d0377be84ca/src/azure-cli-core/azure/cli/core/extension/operations.py#L18

but it may be too heavy for MSAL.

Reply:
Indeed. In fact, `msal-extensions` used to also use `packaging.version` but have since removed it in `https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/125`.